src/rewriters/FieldRewriter.ts

```diff
@@ -0,0 +1,133 @@
+import { ArgumentNode, ASTNode, FieldNode, Kind, SelectionSetNode } from 'graphql';
+import { astArgVarNode, NodeAndVarDefs } from '../ast';
+import Rewriter, { RewriterOpts, Variables } from './Rewriter';
+
+interface FieldRewriterOpts extends RewriterOpts {
+  newFieldName?: string;
+  arguments?: string[];
+  objectFieldName?: string;
+}
+
+/**
+ * More generic version of ScalarFieldToObjectField rewriter
+ */
+class FieldRewriter extends Rewriter {
```

Comment:
This looks like it's doing the same thing as `ScalarFieldToObjectFieldRewriter`, but with the option of rewriting the field name. If that's correct, would it make sense to add a `FieldNameRewriter` instead which can rename a field, and then that could be used in series with the `ScalarFieldToObjectFieldRewriter` rather than trying to duplicate functionality? I'm worried that trying to put too much functionality into a single rewriter will make it confusing what that rewriter is doing.

Reply:
So the FieldRewriter not only renames the field, but also supports tapping into its arguments. I guess, if we follow the approach of having more limited rewriters, we'd need two additional ones: `FieldNameRewriter` and `FieldArgsRewriter`.

Reply:
There's already `FieldArgNameRewriter` and `FieldArgTypeRewriter` for rewriting args. Would anything else besides those be needed?

Reply:
So, adding arguments to a Field that has no initial arguments is missing, also argument value coercion I think is also lacking.

Reply:
`FieldArgTypeRewriter` should be able to do argument value coercion. What's the use case for adding an arg to a field? adding an argument to a field isn't a breaking change

Reply:
You're right FieldArgTypeRewriter  has argument coercion, but it needs to be extended to support coercing arg values when they dont come as variables. 

About the other thing, It shouldn't be a breaking change to add args to a field with no initial args, as its optional. Maybe my use case is somewhat specific, but I used to have a resolver that based on a scalar field it made a totalCount aggregation returning the total number of results. Now I dont have that resolver anymore so I need to convert that scalar field into a object field but also passing a filter argument _(there's a test for this in the rewriteField.test.ts file called 'rewrites a scalar field to be a renamed object field with variable arguments and with 1 scalar subfield')._

Reply:
aah interesting I see what you mean about maybe needing to add an argument to make an old query valid in a new schema. That is a use-case that none of the current rewriters handle currently. Philosophically this library has favored small targeted rewriters for each use-case rather than 1 big one that does most things, but maybe this is a good change actually. For adding arguments, what do you think about making it a map instead of an array? ex `arguments: { arg1: "$arg1" }` instead of `arguments: ['arg1']` to give more control over what the argument value is set to? I can imagine a situation where you might want to set it to a constant or something, like `arguments: { perPage: 1 }`.

Reply:
Yes totally, that sounds even better